Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/revert.py

```diff
@@ -94,65 +94,73 @@ def get_issue_info(cls, args, repository, **kwargs):
+        # Retrieve information for commits to be reverted.
+        for c in args.commit:
+            try:
+                commit = repository.find(c, include_log=True)
+            except (local.Scm.Exception, ValueError) as exception:
+                # ValueErrors and Scm exceptions usually contain enough information to be displayed
+                # to the user as an error
+                sys.stderr.write('Could not find "{}"'.format(c) + '\n')
+                return 1
+            commits_to_revert.append(commit.hash)
+            bug_urls = []
+            commit_title = None
+            for line in commit.message.splitlines():
+                if not commit_title:
+                    commit_title = line
+                tracker = Tracker.from_string(line)
+                if tracker:
+                    bug_urls.append(line)
+
+            reverted_changeset += '\n{}\n'.format(commit_title)
+            reverted_changeset += '\n'.join(bug_urls)
+            if commit.identifier and commit.branch:
+                commit_identifiers.append('{}@{}'.format(commit.identifier, commit.branch))
+                reverted_changeset += '\nhttps://commits.webkit.org/{}@{}\n'.format(commit.identifier, commit.branch)
+            else:
+                sys.stderr.write('Could not find "{}"'.format(', '.join(args.commit)) + '\n')
+                return 1
+
+        # Retrieve information for the issue tracking the revert.
+        # TODO: Get radar?
         revert_issue = Tracker.from_string(args.issue)
         if not revert_issue:
             sys.stderr.write('Could not find issue {}'.format(revert_issue))
-
-        env = os.environ
-        env['COMMIT_MESSAGE_TITLE'] = cls.REVERT_TITLE_TEMPLATE.format(commit, commit_title)
-        env['COMMIT_MESSAGE_REVERT'] = '{}\n\n{}\n\nReverted changeset:\n\n{}'.format(revert_issue.link, revert_issue.title, commit_title)
-        env['COMMIT_MESSAGE_BUG'] = '\n'.join(bug_urls)
-
-        if commit.identifier and commit.branch:
-            env['COMMIT_MESSAGE_BUG'] += '\nhttps://commits.webkit.org/{}@{}'.format(commit.identifier, commit.branch)
-        else:
-            sys.stderr.write('Could not find "{}"'.format(args.commit) + '\n')
             return 1
-
-        return commit, commit_title, bug_urls
+        env = os.environ
+        env['COMMIT_MESSAGE_TITLE'] = cls.REVERT_TITLE_TEMPLATE.format(', '.join(commit_identifiers))
+        env['COMMIT_MESSAGE_REVERT'] = '{}\n\n{}\n\nReverted changeset:\n{}'.format(revert_issue.link, revert_issue.title, reverted_changeset)
+        return commit_identifiers, commits_to_revert
 
     @classmethod
-    def revert_commit(cls, args, repository, issue, commit, commit_title, bug_urls, **kwargs):
-        result = run([repository.executable(), 'revert', '--no-commit'] + [commit.hash], cwd=repository.root_path, capture_output=True)
+    def revert_commit(cls, args, repository, issue, commit_identifiers, commits_to_revert, **kwargs):
```

Comment:
If you just passed `commits`, you could print commit objects. This is probably best anyways, while WebKit uses identifiers, this script should work in repositories which don't.